### PR TITLE
docs: v0.3.10 release

### PR DESCRIPTION
**Pull request by charles-cooper**
### Commit message

Commit message for the final, squashed PR. (Optional, but reviewers will appreciate it! Please see [our commit message style guide](../../blob/master/docs/style-guide.rst#best-practices-1) for what we would ideally like to see in a commit message.)

### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

**Review by fubuloubu — APPROVED** [review]

File: docs/release-notes.rst [hunk 1]
```diff
@@ -14,14 +14,10 @@ Release Notes
...
-v0.3.10rc1
-**********
+v0.3.10 ("Black Adder")
+***********************
 
-Date released: 2023-09-06
+Date released: 2023-10-03
 =========================
 
 v0.3.10 is a performance focused release. It adds a ``codesize`` optimization mode (`#3493 <https://github.com/vyperlang/vyper/pull/3493>`_), adds new vyper-specific ``#pragma`` directives  (`#3493 <https://github.com/vyperlang/vyper/pull/3493>`_), uses Cancun's ``MCOPY`` opcode for some compiler generated code (`#3483 <https://github.com/vyperlang/vyper/pull/3483>`_), and generates selector tables which now feature O(1) performance (`#3496 <https://github.com/vyperlang/vyper/pull/3496>`_).
```
Comment (fubuloubu) [inline docs/release-notes.rst:23]: Is this true still?

**Review by fubuloubu — APPROVED** [review]

File: docs/release-notes.rst (hunk 1 above)
Comment (charles-cooper) [reply docs/release-notes.rst:23]: which -- that it's a performance focused release?

**Comment by charles-cooper** [comment]
note: the fuzz failure in https://github.com/vyperlang/vyper/actions/runs/6394711626/job/17356745080?pr=3629 is not a compilation issue but it's an issue in test construction. the test suite fix is being worked on in https://github.com/vyperlang/vyper/pull/3633

File: docs/release-notes.rst (hunk 1 above)
Comment (fubuloubu) [reply docs/release-notes.rst:23]: lots of bug fixes now

Comment (charles-cooper) [reply docs/release-notes.rst:23]: i'd say still performance focused. there are some bug fixes ofc but there are some in every release, that doesn't make this not performance focused.

**Pull request merged**